src/Symfony/Component/Form/PreloadedExtension.php

```diff
@@ -44,9 +44,16 @@ class PreloadedExtension implements FormExtensionInterface
      */
     public function __construct(array $types, array $typeExtensions, FormTypeGuesserInterface $typeGuesser = null)
     {
-        $this->types = $types;
         $this->typeExtensions = $typeExtensions;
         $this->typeGuesser = $typeGuesser;
+
+        foreach ($types as $name => $type) {
+            // Up to Symfony 2.8, types were identified by their names
+            $this->types[$name] = $type;
```

Comment:
you should check about ``$name`` being an integer to avoid registering in this case for people using the 3.0 syntax and giving only an array of types (and btw, why are we forcing to pass names as keys when the type object passed as value already has the info?)